### PR TITLE
Support for comparing against a null value using EQ and NEQ methods

### PR DESCRIPTION
**Pull request by mkruk-u2**

### CODE REVIEW

**Comment by mkruk-u2** [comment]
Do we want to support PHP 5.5 still?

**Comment by yakobe** [comment]
> Do we want to support PHP 5.5 still?

I guess we should unless we bump versions

File: DataSource/Driver/Doctrine/QueryBuilder/Filterer.php [hunk 1]
```diff
@@ -144,6 +145,22 @@ protected function buildCondition(FilterCondition $filterCondition, $token, Quer
         $identifier = $this->uniqueNameToQueryFieldMap[$filterCondition->getFieldName()];
         $value = $filterCondition->getValue();
 
+        if (null === $value) {
```
Comment (mkruk-u2) [inline DataSource/Driver/Doctrine/QueryBuilder/Filterer.php]: Ugly.
Refactor it.

File: DataSource/Driver/Doctrine/QueryBuilder/Filterer.php [hunk 2]
```diff
@@ -144,6 +119,10 @@ protected function buildCondition(FilterCondition $filterCondition, $token, Quer
...
+            return $this->buildEmptinessCondition($filterCondition, $identifier, $queryBuilder);
```
Comment (yakobe) [inline DataSource/Driver/Doctrine/QueryBuilder/Filterer.php]: buildExpressionForNullValue?

File: DataSource/Driver/Doctrine/QueryBuilder/Filterer.php [hunk 3]
```diff
@@ -170,21 +149,7 @@ protected function buildCondition(FilterCondition $filterCondition, $token, Quer
...
-                    // to emulate this without actually using an invalid empty array in the IN statement.
-                    return '1=2';
-                }
-
-                if ($filterMethod === FilterCondition::METHOD_IN) {
-                    return $queryBuilder->expr()->in($identifier, $token);
-                } else {
-                    return $queryBuilder->expr()->notIn($identifier, $token);
-                }
+                return $this->buildSetInclusionCondition($filterCondition, $identifier, $token, $queryBuilder);
```
Comment (yakobe) [inline DataSource/Driver/Doctrine/QueryBuilder/Filterer.php]: buildExpressionForCollection

**Comment by yakobe** [comment]
Looks good. We could maybe improve a couple of private function names, but they are not critical. So... :shipit: when you are ready

**Pull request merged**